internal/template/function.go

```
@@ -82,3 +84,50 @@ func resourceRequests(pods corev1.PodList, weights string) (float64, error) {
 	}
 	return totalResources, nil
 }
+
+func cpuUtilization(labelArgs ...string) (string, error) {
+
+	cpuUtilizationQueryTemplate := `
+scalar(
+  sum(
+    max(container_cpu_usage_seconds_total{container="", image=""}) by (pod)
+    *
+    on (pod) group_left kube_pod_labels{{ . }}
+  )
+  /
+  sum(
+    sum_over_time(kube_pod_container_resource_limits_cpu_cores[1h:1s])
+    *
+    on (pod) group_left kube_pod_labels{{ . }}
+  )
+)`
+
+	tmpl, err := template.New("query").Parse(cpuUtilizationQueryTemplate)
+	if err != nil {
+		return "", err
+	}
+
+	var labels []string
+	for _, label := range strings.Split(strings.Join(labelArgs, ","), ",") {
+		kvpair := strings.Split(label, "=")
+		if len(kvpair) != 2 {
+			// Should we continue or hard fail on invalid labels?
```

I would fail the metric scrape if the labels are invalid